### ISSUE TITLE
Override validate fn with :validate-fn

### ISSUE THREAD

**Issue by pate** [Day8/re-com#41]
Please let me validate input-text input with my own fn. Right now I have to do it in on-change, but would like to use re-com's validation statuses.

**Comment by mike-thompson-day8** [comment]
I'm afraid I don't follow.  Can you provide more detail about your request please.  Is this related to the #43 which you raised?

**Comment by pate** [comment]
This issue is distinct from #43. By validation, I mean validating entered text, not restricting input. So, for example, a user may enter a number, but only certain numbers would be considered valid.

E.g. I want to do a Luhn check on an IMEI number or a credit card number, which cannot be done with a regular expression. Instead of preventing entry, I want to give the user feedback about invalid input.

**Comment by mike-thompson-day8** [comment]
As per #43 you can get character by character callbacks via `on-change` calls.  Each time you update the model or not.

**Comment by mike-thompson-day8** [comment]
For example, the following (untested) code will never allow the user to enter an 'e' character:
```
(defn test
    [] 
    (let [model   (reagent/ratom  "world")]
      (fn []
           [input-text 
                 :model   model 
                 :change-on-blur? false
                 :on-change  (fn [new-value]   (if-not  (some #(= \e %) new-value)  (reset! model new-value))])))
```

**Comment by pate** [comment]
Thanks, that attribute solves my problem :). I saw the `:change-on-blur?` attribute, but it was not obvious to me that false would result in on-change firing on-key-up. I tried specifying `:on-key-up` fn, but that was not allowed. May I suggest considering `{:fire-change :on-blur/:on-key-up}` for future APIs.

**Comment by mike-thompson-day8** [comment]
Yep, there was a reason I suggested this was a duplicate of #43.
